recipes/libsystemd/all/conanfile.py

```diff
@@ -162,8 +166,10 @@ def _patch_sources(self):
         apply_conandata_patches(self)
 
         meson_build = os.path.join(self.source_folder, "meson.build")
-        replace_in_file(self, meson_build, "@CONAN_SRC_REL_PATH@",
-                        "'../{}'".format(os.path.basename(self.source_folder)))
+        replace_in_file(self, meson_build, "@CONAN_SRC_REL_PATH@", f"'../{self.source_path.name}'")
+        # Fix an invalid path separator, which breaks with newer Meson versions
```

Comment:
Has this been fixed upstream?

Reply:
No. I opened a PR: https://github.com/systemd/systemd/pull/30359

Reply:
I have, of course, but was not familiar with the way systemd handles it. My "fix" is most certainly incorrect, apologies.

The reason I assumed it's a path separator issue was due to the decompressed source archive incorrectly splitting these files up into subdirs. It appears to be a Conan bug/limitation.